libtiledbsoma/src/reindexer/reindexer.h

```diff
@@ -74,12 +74,8 @@ class IntIndexer {
         lookup(keys.data(), results.data(), keys.size());
     }
     IntIndexer(){};
-    /**
-     * Constructor with the same arguments as map_locations
-     */
-    IntIndexer(const int64_t* keys, int size, int threads);
-    IntIndexer(const std::vector<int64_t>& keys, int threads)
-        : IntIndexer(keys.data(), keys.size(), threads) {
+    IntIndexer(std::shared_ptr<tiledbsoma::SOMAContext> context) {
+        context_ = context;
```

Comment:
I think that we should always have a context.

```
    IntIndexer() : context_(make_shared<SOMAContext>()){}; // or IntIndexer() = delete;?
    IntIndexer(std::shared_ptr<tiledbsoma::SOMAContext> context) : context_(context) {};
```

Reply:
I wouldn't create a new context in IntIndexer but agree with other suggestions.